src/Command/AbstractCustomCommand.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Command;
+
+use App\Service\Command\CommandCollection;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * AbstractCustomCommand is the abstract class extended by all custom commands that can be chained.
+ */
+abstract class AbstractCustomCommand extends Command
```

Comment:
It's better to handle chaining and logging outside of the command classes, e.g. in the event listener subscribed to console commands events